src/features/lightspeed/utils/data.ts

```diff
@@ -503,3 +492,30 @@ function shouldTriggerMultiTaskSuggestionForPlaybook(
 
 export const matchKeyword = (keywordsRegex: RegExp[], line: string) =>
   keywordsRegex.some((keywordRegex) => keywordRegex.test(line));
+
+export async function getRoleYamlFiles(
+  rolePath: string,
+): Promise<GenerationListEntry[]> {
+  const files = [] as GenerationListEntry[];
+  const directories = ["defaults", "tasks"];
+  directories.forEach(async (dir) => {
+    const dirPath = `${rolePath}/${dir}`;
+    if (fs.existsSync(dirPath)) {
+      const yamlFiles = await fs
+        .readdirSync(dirPath)
+        .filter((file) => /\.(yml|yaml)$/.test(file));
+      yamlFiles.forEach((file) => {
+        const fileContents = readVarFiles(`${dirPath}/${file}`);
+        if (fileContents) {
+          files.push({
+            path: `${dirPath}/${file}`,
+            file_type: dir.slice(0, -1) as "default" | "task",
```

Comment:
We already have the `GenerationFileType` enum that we can rename and reuse here.